fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -365,6 +367,26 @@ public void updateUniquenessViolations(@Nonnull Tuple valueKey, @Nonnull Tuple p
         return keyValues.map(kv -> unpackKeyValue(uniquenessViolationsSubspace, kv));
     }
 
+    /**
+     * Validate entries in the index. It scans the index and checks if the record associated with each index entry exists.
+     * @param scanType the {@link IndexScanType type} of scan to perform
+     * @param range the range to validate
+     * @param continuation any continuation from a previous validation invocation
+     * @param scanProperties skip, limit and other properties of the validation
+     * @return a cursor over index entries that have no associated records.
+     */
+    @Nonnull
+    public RecordCursor<IndexEntry> validateOrphanEntries(@Nonnull IndexScanType scanType,
+                                                          @Nonnull TupleRange range,
+                                                          @Nullable byte[] continuation,
+                                                          @Nonnull ScanProperties scanProperties) {
+        return scan(scanType, range, continuation, scanProperties).filterAsync(
+                indexEntry -> state.store
+                        .hasIndexEntryRecord(state.index, indexEntry, scanProperties.getExecuteProperties().getIsolationLevel())
+                        .thenApply(has -> !has),
+                FDBRecordStore.DEFAULT_PIPELINE_SIZE);
```

Comment:
Will this validation work on all sub-classes of `StandardIndexMaintainer` in Record Layer?  For example, what happens with the rank index or count index?  There should probably be tests for this and it may be worth consulting with @MMcM about whether or not issues should be opened to piggyback on this infrastructure to do additional validation checks on the other index types.

Reply:
Only support orphan validation in `ValueIndexMaintainer` for now as what we discussed offline.